test/taproot-check-test.js

```diff
@@ -251,3 +259,124 @@ describe('Taproot Check', function() {
     }
   });
 });
+
+describe('Helper Functions', () => {
+  it('taprootTreeHelper should produce tree root', () => {
+    /*
+      The outputs for the taproot tree helper function are compared against the results
+      produced by the bitcoin core implementation.
+    */
+
+    const b = new Script({ raw: Buffer.from('b') });
+    const c = new Script({ raw: Buffer.from('c') });
+    const d = new Script({ raw: Buffer.from('d') });
+    const f = new Script({ raw: Buffer.from('f') });
+    const g = new Script({ raw: Buffer.from('g') });
+
+    assert.equal(
+      Taproot.taprootTreeHelper([]),
+      null,
+      'empty'
+    );
+
```

Comment:
I think the whole following section could benefit from a few helpers to clean up:

```diff
diff --git a/test/taproot-check-test.js b/test/taproot-check-test.js
index a89f6b825..ecf9d383e 100644
--- a/test/taproot-check-test.js
+++ b/test/taproot-check-test.js
@@ -273,6 +273,14 @@ describe('Helper Functions', () => {
     const f = new Script({ raw: Buffer.from('f') });
     const g = new Script({ raw: Buffer.from('g') });
 
+    function buf(hex) {
+      return Buffer.from(hex, 'hex');
+    }
+
+    function tap(script, version = common.LEAF_VERSION_TAPSCRIPT) {
+      return new Taproot.TapLeaf(script, version);
+    }
+
     assert.equal(
       Taproot.taprootTreeHelper([]),
       null,
@@ -280,50 +288,50 @@ describe('Helper Functions', () => {
     );
 
     assert.bufferEqual(
-      Taproot.taprootTreeHelper([new Taproot.TapLeaf(b, common.LEAF_VERSION_TAPSCRIPT)]),
-      Buffer.from('c1b5bd5af873b3cf6e5a90ed7dfa03da09ad4c4f61aedb4357c87f13244d0d44', 'hex'),
+      Taproot.taprootTreeHelper([tap(b)]),
+      buf('c1b5bd5af873b3cf6e5a90ed7dfa03da09ad4c4f61aedb4357c87f13244d0d44'),
       '1 leaf'
     );
 
     assert.bufferEqual(
-      Taproot.taprootTreeHelper([new Taproot.TapLeaf(b, 194)]),
-      Buffer.from('06abb1a7f74bbb1030c8385a757352d4bde36743c4ba1eab734c9b8441e10b93', 'hex'),
+      Taproot.taprootTreeHelper([tap(b, 194)]),
+      buf('06abb1a7f74bbb1030c8385a757352d4bde36743c4ba1eab734c9b8441e10b93'),
       'diff leaf version'
     );
 
     assert.bufferEqual(
-      Taproot.taprootTreeHelper([new Taproot.TapLeaf(c, common.LEAF_VERSION_TAPSCRIPT)]),
-      Buffer.from('993e66f0dc536073d5a2c5989267bc8762045303b91c027bf33666565a85f270', 'hex'),
+      Taproot.taprootTreeHelper([tap(c)]),
+      buf('993e66f0dc536073d5a2c5989267bc8762045303b91c027bf33666565a85f270'),
       'diff code'
     );
 
     assert.bufferEqual(
-      Taproot.taprootTreeHelper([[[[[new Taproot.TapLeaf(b, common.LEAF_VERSION_TAPSCRIPT)]]]]]),
-      Buffer.from('c1b5bd5af873b3cf6e5a90ed7dfa03da09ad4c4f61aedb4357c87f13244d0d44', 'hex'),
+      Taproot.taprootTreeHelper([[[[[tap(b)]]]]]),
+      buf('c1b5bd5af873b3cf6e5a90ed7dfa03da09ad4c4f61aedb4357c87f13244d0d44'),
       'deep leaf'
     );
 
     assert.bufferEqual(
-      Taproot.taprootTreeHelper([new Taproot.TapLeaf(b, common.LEAF_VERSION_TAPSCRIPT), new Taproot.TapLeaf(b, common.LEAF_VERSION_TAPSCRIPT)]),
-      Buffer.from('12991d5d42735f679bb1a24a4026f4a6e4fbe49612e2b944f26ab93198253912', 'hex'),
+      Taproot.taprootTreeHelper([tap(b), tap(b)]),
+      buf('12991d5d42735f679bb1a24a4026f4a6e4fbe49612e2b944f26ab93198253912'),
       '2 same leaves'
     );
 
     assert.bufferEqual(
-      Taproot.taprootTreeHelper([new Taproot.TapLeaf(b, common.LEAF_VERSION_TAPSCRIPT), new Taproot.TapLeaf(c, common.LEAF_VERSION_TAPSCRIPT)]),
-      Buffer.from('5314984f24ab08113d5790636c6c7fb8a003e60b50e5b600b62e97d04133e4a5', 'hex'),
+      Taproot.taprootTreeHelper([tap(b), tap(c)]),
+      buf('5314984f24ab08113d5790636c6c7fb8a003e60b50e5b600b62e97d04133e4a5'),
       '2 diff leaves'
     );
 
     assert.bufferEqual(
-      Taproot.taprootTreeHelper([new Taproot.TapLeaf(b, common.LEAF_VERSION_TAPSCRIPT), [new Taproot.TapLeaf(c, common.LEAF_VERSION_TAPSCRIPT)], new Taproot.TapLeaf(d, common.LEAF_VERSION_TAPSCRIPT), [new Taproot.TapLeaf(f, common.LEAF_VERSION_TAPSCRIPT), new Taproot.TapLeaf(g, common.LEAF_VERSION_TAPSCRIPT)]]),
-      Buffer.from('7d06227ec4d4dc84ec46a62481f86cea3466d2479184e5dc17145976bf361aef', 'hex'),
+      Taproot.taprootTreeHelper([tap(b), [tap(c)], tap(d), [tap(f), tap(g)]]),
+      buf('7d06227ec4d4dc84ec46a62481f86cea3466d2479184e5dc17145976bf361aef'),
       'multiple levels'
     );
 
     assert.bufferEqual(
-      Taproot.taprootTreeHelper([[new Taproot.TapLeaf(b, common.LEAF_VERSION_TAPSCRIPT), new Taproot.TapLeaf(c, common.LEAF_VERSION_TAPSCRIPT), new Taproot.TapLeaf(d, common.LEAF_VERSION_TAPSCRIPT)], [new Taproot.TapLeaf(f, common.LEAF_VERSION_TAPSCRIPT), new Taproot.TapLeaf(g, common.LEAF_VERSION_TAPSCRIPT)]]),
-      Buffer.from('eab7f3ca183c40faed41641c972acf02cfaa537e124b2f3806b5323b84386426', 'hex'),
+      Taproot.taprootTreeHelper([[tap(b), tap(c), tap(d)], [tap(f), tap(g)]]),
+      buf('eab7f3ca183c40faed41641c972acf02cfaa537e124b2f3806b5323b84386426'),
       'tree structure'
     );
   });

```